### ISSUE TITLE
Map Input for Joy Button is mixed since 3.2.2

### ISSUE THREAD

**Issue by RonanZe** [godotengine/godot#38869]
**Godot version:** 
Godot_v3.2.2-beta2_win64

**OS/device including version:**
Win 8.1

**Issue description:**
the Map Input for Joy Button is mixed since 3.2.2 beta1 and 2.
I'm using a WiiU pro Controller and a DualShock 4 on PC and both match  in previous versions (3.2.1, 3.1 and 3.0) from the buttons placement point of view.

**Comment by akien-mga** [comment]
Are you sure about 3.2.2 beta 1?
There's a bug in 3.2.2 beta 2 #38554 that broke some gamepads on Windows, since fixed, but it was not in beta 1.

**Comment by RonanZe** [comment]
You are right! It's not in beta1. Only the beta2
I convinced myself I was waiting for beta2 to solve this problem :)

**Comment by akien-mga** [comment]
I'll close as duplicate of #38554 then, but please test the upcoming beta 3 (likely out tomorrow) to confirm that it's fixed :)

**Comment by akien-mga** [comment]
Reopening as per https://github.com/godotengine/godot/issues/38554#issuecomment-633212663

**Comment by akien-mga** [comment]
> Still have the issue in 3.2.2-beta3

What device(s) can you reproduce the issue with? Can you retrieve their GUID with `Input.get_joy_guid(0)` (adjust the device ID if you have more than one plugged in).

Can you then verify if https://www.generalarcade.com/gamepadtool/ has correct mappings for your device(s), and if not, generate new ones that would work?

**Comment by RonanZe** [comment]
print(Input.get_joy_name(0)) --> "Mayflash WiiU Pro Game Controller Adapter"
print(Input.get_joy_guid(0)) --> "79000018000000000000504944564944"

The SDL2 Gamepad Tool give me a correct map (except for the "home" button that don't seems to be supported in hardware by the Mayflash adapter)

**Comment by akien-mga** [comment]
Thanks, it seems indeed that this GUID was part of the old `gamecontrollerdb_204.txt` (SDL 2.0.4 and earlier) from https://github.com/gabomdq/SDL_GameControllerDB, which we removed in #38292.

My understanding is that all gamepads should be mapped in the currently maintained `gamecontrollerdb.txt` at https://github.com/gabomdq/SDL_GameControllerDB, so we should likely make sure that mappings for this device are re-added to the new DB.

Could you create a PR at https://github.com/gabomdq/SDL_GameControllerDB to add mappings for your device? There are instructions at https://github.com/gabomdq/SDL_GameControllerDB#create-new-mappings, SDL2 Gamepad Tool should let you generate the mapping.

For the reference, the one which was used in `gamecontrollerdb_204.txt` was:
```
79000018000000000000504944564944,Mayflash WiiU Pro Game Controller Adapter (DInput),a:b1,b:b2,back:b8,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,leftshoulder:b4,leftstick:b10,lefttrigger:b6,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b11,righttrigger:b7,rightx:a2,righty:a3,start:b9,x:b0,y:b3,platform:Windows,
```
But the format for the new `gamecontrollerdb.txt` should change a bit.

**Comment by hoontee** [comment]
The database already contains a "Mayflash WiiU Pro Game Controller Adapter", but with the only difference being a different GUID. I'll submit a PR to add the other one.

**Comment by RonanZe** [comment]
Are you sure it's not an ID problem? Maybe there are not compute the same way in the master gabomdq/SDL_GameControllerDB

Because I see the gamepad in the "gamecontrollerdb.txt" of the master branch but with this ID instead: 03000000790000000018000000000000

I have to check with my DS3 controller

**Comment by akien-mga** [comment]
There can be similar devices (same nale and mapping) with different GUIDs a priori.

**Comment by akien-mga** [comment]
@RonanZe Can you try to generate new mappings for this device using one of the tools listed at https://github.com/gabomdq/SDL_GameControllerDB#mapping-tools ?

Apparently `79000018000000000000504944564944` is the old GUID format, so that should mean that current tools should give a new GUID for your device.

**Comment by RonanZe** [comment]
@akien-mga 

The SDL2 Gamepad Tool give me this:
Searching gamepads...
Found 4 gamepad(s):
"Mayflash WiiU Pro Game Controller Adapter (DInput)", 03000000790000000018000000000000 (mapping available)
"Mayflash WiiU Pro Game Controller Adapter (DInput)", 03000000790000000018000000000000 (mapping available)
"Mayflash WiiU Pro Game Controller Adapter (DInput)", 03000000790000000018000000000000 (mapping available)
"Mayflash WiiU Pro Game Controller Adapter (DInput)", 03000000790000000018000000000000 (mapping available)

It's matching the ID in gamecontrollerdb.txt found here https://github.com/gabomdq/SDL_GameControllerDB

**Comment by RonanZe** [comment]
Godot is not giving the right GUID (get_joy_guid(0) in Godot 3.2.2-beta == 79000018000000000000504944564944)

**Comment by akien-mga** [comment]
> Godot is not giving the right GUID (get_joy_guid(0) in Godot 3.2.2-beta == 79000018000000000000504944564944)

According to https://github.com/gabomdq/SDL_GameControllerDB/pull/348#issuecomment-633309848 it seems SDL changed the way they generate GUIDs, and we might need to sync with them to properly support the new binding formats.

CC @madmiraal 

Notably this might need an update: https://github.com/godotengine/godot/blob/11d6c0f20dd5bea3b21a90e144ddb75908343280/platform/windows/joypad_windows.cpp#L166-L171

**Comment by akien-mga** [comment]
The SDL commit that changed the GUID format in 2.0.6+ seems to be: http://hg.libsdl.org/SDL/rev/9da2701eeb4a

It notably changes the handling of the DInput GUID adding a prefix to distinguish between USB and Bluetooth devices.

I can have a look at writing a fix but I think you'd be much more efficient than I would be @madmiraal, so tell me if you're up for looking into it :)
I haven't reviewed if GUIDs changed for other platforms, but I guess they did comparing `gamecontrollerdb.txt` (most starting with `03000000` or similar) and `gamecontrollerdb_204.txt` and `gamecontrollerdb_205.txt` (easy to find back in `3.1` branch).

If we want a quick fix for 3.2.2 I could revert #38292.

**Comment by madmiraal** [comment]
I'll have a look.
